plugins/processors/ec2tagger/ec2tagger.go

```diff
@@ -309,25 +242,22 @@ func (t *Tagger) Init() error {
 			t.metadataLookup.instanceId = true
 		case mdKeyImageId:
 			t.metadataLookup.imageId = true
-		case mdKeyInstaneType:
+		case mdKeyInstanceType:
 			t.metadataLookup.instanceType = true
 		default:
-			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s", tag)
+			t.Log.Errorf("ec2tagger: Unsupported EC2 Metadata key: %s.", tag)
 		}
 	}
 
-	t.Log.Infof("ec2tagger: Check ec2 metadata")
-	if !t.ec2metadata.Available() {
-		msg := "ec2tagger: Unable to retrieve InstanceId. This plugin must only be used on an EC2 instance"
-		t.Log.Errorf(msg)
-		return errors.New(msg)
-	}
+	t.Log.Infof("ec2tagger: Check EC2 Metadata.")
 
-	doc, err := t.ec2metadata.GetInstanceIdentityDocument()
-	if nil != err {
-		msg := fmt.Sprintf("ec2tagger: Unable to retrieve InstanceId : %+v", err.Error())
-		t.Log.Errorf(msg)
-		return errors.New(msg)
+	doc, err := t.ec2MetadataProvider().GetInstanceIdentityDocument()
+	if err != nil {
```

Comment:
We do not know 100% this is timeout imo.

Reply:
Agree on the HTTP Error Request. However, it will always return timeout even though the request returns a different error. Timeout is only a consolidate message for most of the errors unless we want to handle every single error.